swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/AmazonAutoScalingGroup.kt

```diff
@@ -16,31 +16,69 @@
 
 package com.netflix.spinnaker.swabbie.aws.autoscalinggroups
 
+import com.amazonaws.services.autoscaling.model.SuspendedProcess
 import com.fasterxml.jackson.annotation.JsonTypeName
+import com.netflix.spinnaker.swabbie.Dates
 import com.netflix.spinnaker.swabbie.aws.model.AmazonResource
 import com.netflix.spinnaker.swabbie.model.AWS
 import com.netflix.spinnaker.swabbie.model.SERVER_GROUP
+import java.time.Instant
 import java.time.LocalDateTime
 import java.time.ZoneId
-import java.time.Instant
 
 @JsonTypeName("amazonAutoScalingGroup")
 data class AmazonAutoScalingGroup(
   val autoScalingGroupName: String,
   val instances: List<Map<String, Any>>?,
   val loadBalancerNames: List<String>?,
+  var suspendedProcesses: List<SuspendedProcess>? = null,
   private val createdTime: Long,
   override val resourceId: String = autoScalingGroupName,
   override val resourceType: String = SERVER_GROUP,
   override val cloudProvider: String = AWS,
   override val name: String = autoScalingGroupName,
   private val creationDate: String? = LocalDateTime.ofInstant(Instant.ofEpochMilli(createdTime), ZoneId.systemDefault()).toString()
 ) : AmazonResource(creationDate) {
+  private val suspensionReasonDateRegex = "(\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2})".toRegex()
+
   override fun equals(other: Any?): Boolean {
     return super.equals(other)
   }
 
   override fun hashCode(): Int {
     return super.hashCode()
   }
+
+  fun isOutOfLoadBalancer(): Boolean {
+    return getAddToLoadBalancerProcess() != null
+  }
+
+  private fun getAddToLoadBalancerProcess(): SuspendedProcess? {
+    return suspendedProcesses?.find {
+      it.processName == "AddToLoadBalancer"
+    }
+  }
+
+  private fun getLoadBalancerSuspensionReason(): String? {
+    return getAddToLoadBalancerProcess()?.suspensionReason
+  }
+
+  fun isDisabled(): Boolean {
+    return (details[IS_DISABLED] != null && details[IS_DISABLED] == true)
+  }
+
+  fun hasInstances(): Boolean {
+    return (details[HAS_INSTANCES] != null && details[HAS_INSTANCES] == true)
+  }
+
+  // TODO :aravindd refactor this method
+  fun disabledTime(): LocalDateTime? {
+    if (!isOutOfLoadBalancer()) {
+      return null
+    }
+
+    val suspensionReason = getLoadBalancerSuspensionReason()!!
```

Comment:
Are you 100% certain this will never be null? I'm not. Better to handle the null case by throwing an exception or logging rather than asserting this will never be null with `!!`